core/src/main/java/org/wso2/testgrid/core/phase/DeployPhase.java

```diff
@@ -117,25 +113,26 @@ private DeploymentCreationResult createDeployment()
         Path infraOutFilePath = DataBucketsHelper.getOutputLocation(getTestPlan())
                 .resolve(DataBucketsHelper.INFRA_OUT_FILE);
         try {
-            printMessage("\t\t Creating deployment: " + getTestPlan().getDeploymentConfig()
-                    .getDeploymentPatterns().get(0).getName());
+            DeploymentCreationResult result = new DeploymentCreationResult();
+            for(Script script: getTestPlan().getDeploymentConfig().getFirstDeploymentPattern().getScripts()) {
+                printMessage("\t\t Creating deployment: " + script.getName());
+
+                if (!infrastructureProvisionResult.isSuccess()) {
```

Comment:
I don't understand this particular logic. Why is this inside a for-loop?

Reply:
This is because in the deploy phase different types of scripts are run. i.e- in kubernetes deployer shell scripts as well as kubernetes scripts are fun. hence the scripts are run through for loop

Reply:
Ok. But, do we need to check this condition `if (!infrastructureProvisionResult.isSuccess()) {` inside a for-loop of deployment phase?